src/tuwien/auto/calimero/tools/Discover.java

```diff
@@ -411,8 +412,10 @@ else if (Main.isOption(arg, "search", "s"))
 				options.put("search", null);
 			else if (Main.isOption(arg, "unicast", "u"))
 				options.put("mcastResponse", Boolean.FALSE);
-			else if (arg.equals("sd"))
+			else if (arg.equals("sd")
+				|| Main.isOption(arg, "searchWithDescription", null)) {
```

Comment:
The reason I didn't document `sd`as command was because I thought that `search` is actually one of two mandatory commands (and not an option), and the search _with description_ would have an option modifying the search command. If you are also in favor of this behavior, then I would shorten the option to `--withDescription` (and an invocation would look like `Discover search --withDesc...`).

Reply:
Ah, that makes sense. I did not grasp the idea that "sd" was supposed to be an argument to "--search", I did always use it without it. So that's why I was wondering why the long option was not implemented.
Obviously the fix needs to be different than, if "sd" is given it must be preceded by "--search". If you think it's worth I will implement it that way, otherwise I would rather drop this patch.

Reply:
I did implement it anyway, see the other pull request.

Reply:
I think my comment caused more confusion than clarity :D

I was trying to make the distinction between _commands_ (no hyphen prefix) and _options_ (`-` `--`) arguments, and pondered on my former (somewhat weird) implementation of `--search` instead of just `search` (like i did with the undocumented `sd` or commands in other tools).
So I was suggesting to maybe change/document the option `--searchWithDescription` as `--withDescription` and eventually have the following syntax, e.g.:
* `search`
* `search --timeout 2`
* `search --withDescription --timeout 4`

So `sd` is the shortcut to `search --withDescription`.

(All of this is a suggestion; if you think it makes user invocation worse, I will leave it as is now)

Reply:
I took your pr and continued it, so that i can use the new commands for the 2.4 branch. I will close the other related pr. Thanks!